flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/extraction/ExtractionUtils.java

```diff
@@ -936,10 +947,11 @@ public void visitLocalVariable(
      * @param cls the Class to check, may be null
      * @param toClass the Class to try to assign into, returns false if null
      * @param autoboxing whether to use implicit autoboxing/unboxing between primitives and wrappers
+     * @param strictAutoboxing checks whether null would end up in a primitive type and forbids it
      * @return {@code true} if assignment possible
      */
     public static boolean isAssignable(
-            Class<?> cls, final Class<?> toClass, final boolean autoboxing) {
+            Class<?> cls, final Class<?> toClass, boolean autoboxing, boolean strictAutoboxing) {
```

Comment:
Can we be consistent with `final`? I believe both `autoboxing` and `strictAutoboxing` can be `final`